tests/MiniProfiler.Tests/MiniProfilerTest.cs

```diff
@@ -33,6 +36,102 @@ public void Simple()
         }
 
         [Fact]
+        public void WhenUsingAsyncProvider_SimpleCaseWorking()
+        {
+            MiniProfiler.Settings.ProfilerProvider = new DefaultProfilerProvider();
+            using (GetRequest())
+            {
+                MiniProfiler.Start();
+                IncrementStopwatch(); // 1 ms
+                MiniProfiler.Stop();
+
+                var c = MiniProfiler.Current;
+
+                Assert.NotNull(c);
+                Assert.Equal(StepTimeMilliseconds, c.DurationMilliseconds);
+
+                Assert.NotNull(c.Root);
+                Assert.False(c.Root.HasChildren);
+            }
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Current_WhenAsyncMethodReturns_IsCarried(
+            bool configureAwait
+            )
+        {
+            MiniProfiler.Settings.ProfilerProvider = new DefaultProfilerProvider();
+            using (GetRequest())
+            {
+                MiniProfiler.Start();
+
+                var c = MiniProfiler.Current;
+                await Task.Delay(TimeSpan.FromMilliseconds(1)).ConfigureAwait(configureAwait);
+
+                Assert.NotNull(MiniProfiler.Current);
+                Assert.Equal(c, MiniProfiler.Current);
+            }
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Head_WhenAsyncMethodReturns_IsCarried(
+            bool configureAwait
+            )
+        {
+            MiniProfiler.Settings.ProfilerProvider = new DefaultProfilerProvider();
+            using (GetRequest())
+            {
+                MiniProfiler.Start();
+                var sut = MiniProfiler.Current;
+                var head = sut.Head;
+
+                await Task.Delay(TimeSpan.FromMilliseconds(1)).ConfigureAwait(configureAwait);
+
+                Assert.NotNull(sut.Head);
+                Assert.Equal(head, sut.Head);
+            }
+        }
+
+        [Fact]
+        public void Head_WhenMultipleTasksSpawned_EachSetsItsOwnHead()
```

Comment:
this test is a demonstration of the scenario i'm trying to support when one thread spawns two tasks and wait for their completion. In this scenario I would expect the tasks's Timings to be children of the main thread's Timing instance